### PR TITLE
#186810782 : replace print by logger

### PR DESCRIPTION
**Pull request by HzANut**

### CODE REVIEW

File: moesifasgi/update_companies.py [hunk 1]
```diff
@@ -11,91 +14,92 @@ def __init__(self):
...
-                            print("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
+                            logger.error("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
                         if DEBUG:
-                            print("Error while updating companies, with status code:")
-                            print(inst.response_code)
+                            logger.info("Error while updating companies, with status code: {str(inst.response_code)}")
                 else:
-                    print('To update companies, an company_id field is required')
+                    if DEBUG:
+                        logger.info('To update companies, an company_id field is required')
```
Comment (praves77) [inline moesifasgi/update_companies.py]: this is warning or error and don't need `debug` here.

File: moesifasgi/update_companies.py [hunk 2]
```diff
@@ -11,91 +14,92 @@ def __init__(self):
...
-                            print("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
+                            logger.error("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
                         if DEBUG:
-                            print("Error while updating companies, with status code:")
-                            print(inst.response_code)
+                            logger.info(f"Error while updating companies, with status code: {str(inst.response_code)}")
                 else:
-                    print('To update companies, an company_id field is required')
+                    if DEBUG:
+                        logger.info('To update companies, an company_id field is required')
```
Comment (praves77) [inline moesifasgi/update_companies.py]: same here for `required` field if missing.

File: moesifasgi/update_users.py [hunk 3]
```diff
@@ -11,91 +13,93 @@ def __init__(self):
...
-                            print("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
+                            logger.error("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
                         if DEBUG:
-                            print("Error while updating user, with status code:")
-                            print(inst.response_code)
+                            logger.info(f"Error while updating user, with status code: {str(inst.response_code)}")
                 else:
-                    print('To update an user, an user_id field is required')
+                    if DEBUG:
+                        logger.info('To update an user, an user_id field is required')
```
Comment (praves77) [inline moesifasgi/update_users.py]: same here

File: moesifasgi/update_users.py [hunk 4]
```diff
@@ -11,91 +13,93 @@ def __init__(self):
...
```
Comment (praves77) [inline moesifasgi/update_users.py]: same here

File: moesifasgi/update_users.py [hunk 5]
```diff
@@ -11,91 +13,93 @@ def __init__(self):
...
-                                print("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
+                                logger.error("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
                             if DEBUG:
-                                print("Error while updating user, with status code:")
-                                print(inst.response_code)
+                                logger.info(f"Error while updating user, with status code: {str(inst.response_code)}")
                     else:
-                        print('To update an user, an user_id field is required')
+                        if DEBUG:
+                            logger.info('To update an user, an user_id field is required')
```
Comment (praves77) [inline moesifasgi/update_users.py]: same here

File: moesifasgi/update_users.py [hunk 6]
```diff
@@ -11,91 +13,93 @@ def __init__(self):
...
-                            print("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
+                            logger.error("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
                         if DEBUG:
-                            print("Error while updating users, with status code:")
-                            print(inst.response_code)
+                            logger.info(f"Error while updating users, with status code: {str(inst.response_code)}")
                 else:
-                    print('To update users, an user_id field is required')
+                    if DEBUG:
+                        logger.info('To update users, an user_id field is required')
```
Comment (praves77) [inline moesifasgi/update_users.py]: same here

File: moesifasgi/update_users.py [hunk 7]
```diff
@@ -11,91 +13,93 @@ def __init__(self):
...
                         if 401 <= inst.response_code <= 403:
-                            print("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
+                            logger.error("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
                         if DEBUG:
-                            print("Error while updating users, with status code:")
-                            print(inst.response_code)
+                            logger.info(f"Error while updating users, with status code: {str(inst.response_code)}")
                 else:
-                    print('To update users, an user_id field is required')
+                    if DEBUG:
```
Comment (praves77) [inline moesifasgi/update_users.py]: same

File: moesifasgi/update_users.py [hunk 8]
```diff
@@ -105,14 +109,14 @@ def update_users_batch(cls, user_profiles, api_client, DEBUG):
...
                             if 401 <= inst.response_code <= 403:
-                                print("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
+                                logger.error("Unauthorized access sending event to Moesif. Please check your Appplication Id.")
                             if DEBUG:
-                                print("Error while updating users, with status code:")
-                                print(inst.response_code)
+                                logger.info(f"Error while updating users, with status code: {str(inst.response_code)}")
                     else:
-                        print('To update users, an user_id field is required')
+                        if DEBUG:
```
Comment (praves77) [inline moesifasgi/update_users.py]: same

File: moesifasgi/update_companies.py (hunk 1 above)
Comment (HzANut) [reply moesifasgi/update_companies.py]: changed

File: moesifasgi/update_companies.py (hunk 2 above)
Comment (HzANut) [reply moesifasgi/update_companies.py]: changed

File: moesifasgi/update_users.py (hunk 3 above)
Comment (HzANut) [reply moesifasgi/update_users.py]: changed

File: moesifasgi/update_users.py (hunk 4 above)
Comment (HzANut) [reply moesifasgi/update_users.py]: changed

File: moesifasgi/update_users.py (hunk 5 above)
Comment (HzANut) [reply moesifasgi/update_users.py]: changed

File: moesifasgi/update_users.py (hunk 6 above)
Comment (HzANut) [reply moesifasgi/update_users.py]: changed

File: moesifasgi/update_users.py (hunk 7 above)
Comment (HzANut) [reply moesifasgi/update_users.py]: changed

File: moesifasgi/update_users.py (hunk 8 above)
Comment (HzANut) [reply moesifasgi/update_users.py]: changed

**Review by praves77 — APPROVED** [review]

**Pull request merged**